pkg/storage/image.go

```diff
@@ -555,6 +556,13 @@ func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
 	}
 	domain, remainder := splitDockerDomain(imageName)
 	if domain != "" {
+		registry, err := sysregistriesv2.FindRegistry(nil, domain)
+		if err != nil {
+			return nil, err
+		}
+		if registry != nil && registry.Blocked {
+			return nil, fmt.Errorf("cannot resolve names from registry %q because it's blocked", domain)
+		}
 		// this means the image is already fully qualified
```

Comment:
Could you also add the comment into the domain == ""` case to say that `registry.Blocked` has already been processed earlier?

Although…, (@vrothberg ?), is it logically correct to do `FindRegistry`+`.Blocked` on hostnames only, and not on the full repo paths? That would break the “check `.Blocked` on initialization” optimization — and BTW `FindRegistry` above, should in that case include all of `imageName`, not just `domain`.

Reply:
Good catch, @mtrmac! FindRegistries(...) must be called with `imageName`. At the moment, I might still just work as no `registries.conf` is using the new format yet but it would be a bug once the new format is used.

Reply:
is it working _now_ however if I pass `imageName`?

Reply:
Absolutely (see https://github.com/containers/image/blob/master/pkg/sysregistriesv2/system_registries_v2_test.go#L79).